hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/GenericRecordQueryReader.java

```diff
@@ -246,11 +249,14 @@ private Object readIndexed(InternalGenericRecord record, String path, int index)
     }
 
     private Object readLeaf(InternalGenericRecord record, String path) {
-        if (!record.hasField(path)) {
-            return null;
+        if (record instanceof CompactGenericRecord) {
+            FieldDescriptor fd = ((CompactGenericRecord) record).getFieldDescriptor(path);
```

Comment:
I am looking at the methods like `getString(String fieldName)` and `getString(FieldDescriptor fieldName)`, and the only difference is that you don't call `getFieldDescriptor` in the latter. However, you are calling it here. They look identical to me. So, how is this faster?

Reply:
record.getFieldKind() was getting the FieldDescriptor from schema. And then based on the path we were getting it again.

Now we get the field descriptor at the beginning and reuse it for getting field kind, and reading the field.